lib/filesystem.js

```diff
@@ -101,12 +101,12 @@ class Filesystem {
   insertLink (p) {
     const symlink = fs.readlinkSync(p)
     const parentPath = path.dirname(p)
-    const link = path.relative(fs.realpathSync(this.src), path.join(parentPath, symlink))
+    const link = path.resolve(parentPath, symlink)
     if (link.substr(0, 2) === '..') {
       throw new Error(`${p}: file "${link}" links out of the package`)
     }
     const node = this.searchNodeFromPath(p)
-    node.link = link
+    node.link = path.relative(fs.realpathSync(this.src), link)
     return link
```

Comment:
```ts
const realSrcPath = fs.realpathSync(this.src)
const target = path.join(path.dirname(p), fs.readlinkSync(p))
const relativePath = path.relative(realSrcPath, fs.realpathSync(target))
if (relativePath.startsWith('..')) {
  throw new Error(`${p}: file "${relativePath}" links out of the package`)
}
const node = this.searchNodeFromPath(p)
node.link = path.relative(realSrcPath, target)
return path.relative(realSrcPath, target)
``

Reply:
Can you test the code above? It has been validated locally by me.

Reply:
Yes. It's Ok

Reply:
Cool. Can you apply this modification?

Reply:
My pleasure.